lints/lint_signature_algorithm_weak.go

```diff
@@ -0,0 +1,49 @@
+// lint_signature_algorithm_weak.go
+
+package lints
+
+import (
+	"github.com/zmap/zcrypto/x509"
+	"github.com/zmap/zlint/util"
+	"math/big"
+)
+
+type WeakAlgo struct {
+	// Internal data here
+}
+
+func (l *WeakAlgo) Initialize() error {
+	return nil
+}
+
+func (l *WeakAlgo) CheckApplies(c *x509.Certificate) bool {
+	return true
+}
+
+func (l *WeakAlgo) RunTest(c *x509.Certificate) (ResultStruct, error) {
+	z := big.NewInt(0)
+	if c.SignatureAlgorithm == x509.SHA1WithRSA || c.SignatureAlgorithm == x509.DSAWithSHA1 || c.SignatureAlgorithm == x509.ECDSAWithSHA1 {
+		z.Exp(big.NewInt(2), big.NewInt(64), nil)
+		if c.SerialNumber.Cmp(z) == -1 {
+			return ResultStruct{Result: Warn}, nil
+		}
+	}
+	if c.SignatureAlgorithm == x509.SHA256WithRSA || c.SignatureAlgorithm == x509.SHA384WithRSA || c.SignatureAlgorithm == x509.SHA512WithRSA ||
```

Comment:
Maybe you should create a list of acceptable algorithms. How did you arrive at this list?